azurerm/resource_arm_storage_account.go

```diff
@@ -730,11 +731,46 @@ func resourceArmStorageAccountDelete(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 	name := id.Path["storageAccounts"]
-	resGroup := id.ResourceGroup
+	resourceGroup := id.ResourceGroup
+
+	read, err := client.GetProperties(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(read.Response) {
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Storage Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	virtualNetworkNames := make([]string, 0)
```

Comment:
quick comment explaining why we are locking here might be handy

Reply:
done 👍🏻